cluster/images/etcd/migrate/migrate.go

```diff
@@ -0,0 +1,185 @@
+		Long: `Upgrade or downgrade etcd data across multiple versions to the target version
+
+Given a 'bin-dir' directory of etcd and etcdctl binaries, an etcd 'data-dir' with a 'version.txt' file and
+a target etcd version, this tool will upgrade or downgrade the etcd data from the version specified in
+'version.txt' to the target version.
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			runMigrate()
+		},
+	}
+	opts = migrateOpts{}
+)
+
+type migrateOpts struct {
+	name                 string
+	peerListenUrls       string
+	peerAdvertiseUrls    string
+	binDir               string
+	dataDir              string
+	bundledVersionString string
+	etcdDataPrefix       string
+	ttlKeysDirectory     string
+	initialCluster       string
+	targetVersion        string
+	targetStorage        string
+	etcdServerArgs       string
+}
+
+func main() {
+	flags := migrateCmd.Flags()
+	flags.StringVar(&opts.name, "name", "", "etcd cluster member name. Defaults to etcd-{hostname}")
+	flags.StringVar(&opts.peerListenUrls, "listen-peer-urls", "", "etcd --listen-peer-urls flag, required for HA clusters")
+	flags.StringVar(&opts.peerAdvertiseUrls, "initial-advertise-peer-urls", "", "etcd --initial-advertise-peer-urls flag, required for HA clusters")
+	flags.StringVar(&opts.binDir, "bin-dir", "/usr/local/bin", "directory of etcd and etcdctl binaries, must contain etcd-<version> and etcdctl-<version> for each version listed in bindled-versions")
+	flags.StringVar(&opts.dataDir, "data-dir", "", "etcd data directory of etcd server to migrate")
+	flags.StringVar(&opts.bundledVersionString, "bundled-versions", "", "comma separated list of etcd binary versions present under the bin-dir")
+	flags.StringVar(&opts.etcdDataPrefix, "etcd-data-prefix", "/registry", "etcd key prefix under which all objects are kept")
+	flags.StringVar(&opts.ttlKeysDirectory, "ttl-keys-directory", "", "etcd key prefix under which all keys with TTLs are kept. Defaults to {etcd-data-prefix}/events")
+	flags.StringVar(&opts.initialCluster, "initial-cluster", "", "comma separated list of name=endpoint pairs. Defaults to etcd-{hostname}=http://localhost:2380")
+	flags.StringVar(&opts.targetVersion, "target-version", "", "version of etcd to migrate to. Format must be '<major>.<minor>.<patch>'")
+	flags.StringVar(&opts.targetStorage, "target-storage", "", "storage version of etcd to migrate to, one of: etcd2, etcd3")
+	flags.StringVar(&opts.etcdServerArgs, "etcd-server-extra-args", "", "additional etcd server args for starting etcd servers during migration steps, --peer-* TLS cert flags should be added for etcd clusters with more than 1 member that use mutual TLS for peer communication.")
+	migrateCmd.Execute()
+}
+
+// runMigrate validates the command line flags and starts the migration.
+func runMigrate() {
+	if opts.name == "" {
+		hostname, err := os.Hostname()
+		if err != nil {
+			glog.Errorf("Error while getting hostname to supply default --name: %s", err)
```

Comment:
%v ?

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings seems to suggest that %v is the standard. https://godoc.org/github.com/pkg/errors show either works but being able to extend to more easily switch to %+v seems like a very minor advantage.

Reply:
Good to know. I'll switch over to `%v`.